### PR TITLE
add sustanability support

### PR DESCRIPTION
**Pull request by jiangphcn**
Provide functionalities to get sustainability data. 

- "GetEnergyFromLpar": Get energy consumption for the specified LPAR from historical data. The public document can be found at https://zos-hot-topics.com/2023/z16-l14-sustainability/#get-logical-partition-historical-sustainability-data
- "GetLiveEnergyFromLpar": Get live energy consumption for the specified LPAR. The public information can be found at https://github.com/zhmcclient/python-zhmcclient/blob/master/examples/metrics.py.

### CODE REVIEW

File: sample.go [hunk 1]
```diff
@@ -789,3 +795,31 @@ func UpdateNicProperties(hmcManager zhmcclient.ZhmcAPI) {
 		}
 	}
 }
+
+func GetEnergyFromLpar(hmcManager zhmcclient.ZhmcAPI) {
+
+	lparID := os.Getenv("PAR_ID")
+	lparURI := "api/logical-partitions/" + lparID
```
Comment (huoqifeng) [inline sample.go:802]: I noticed this lparURI does not has `/` at beginning, while same variable at L818 starts as `/`, are they all good?

File: pkg/zhmcclient/lpar.go [hunk 2]
```diff
@@ -42,12 +42,23 @@ type LparAPI interface {
 
 	ListNics(lparURI string) ([]string, int, *HmcError)
 	FetchAsciiConsoleURI(lparURI string, request *AsciiConsoleURIPayload) (*AsciiConsoleURIResponse, int, *HmcError)
+
+	GetEnergyDetailsforLPAR(cpcURI string, props *EnergyRequestPayload) (uint64, int, *HmcError)
```
Comment (huoqifeng) [inline pkg/zhmcclient/lpar.go:46]: looks used different variable name `cpcURI` and `lparURI` here and L630.

**Review by huoqifeng — COMMENTED** [review]
Thank you @jiangphcn added several comments.

File: pkg/zhmcclient/lpar.go (hunk 2 above)
Comment (jiangphcn) [reply pkg/zhmcclient/lpar.go]: yes, we should keep consistency, and use `lparURI` here.

File: sample.go (hunk 1 above)
Comment (jiangphcn) [reply sample.go]: yes, but we should keep consistency and i add"/" at the beginning.

**Review by huoqifeng — APPROVED** [review]
LGTM

**Pull request merged**